x-pack/plugins/observability_solution/synthetics/server/common/inline_to_zip.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import archiver from 'archiver';
+import { MemWritable } from './mem_writable';
+
+function wrapInlineInProject(inlineJourney: string) {
+  return `import { journey, step, expect } from '@elastic/synthetics';
+
+journey('inline', ({ page, context, browser, params, request }) => {
+${inlineJourney}
+});`;
+}
+
+export async function inlineToProjectZip(
+  inlineJourney: string,
+  monitorId: string,
+  logger: Logger
+): Promise<string> {
+  const mWriter = new MemWritable();
+  try {
+    await new Promise<void>((resolve, reject) => {
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      archive.on('error', reject);
+      mWriter.on('close', resolve);
+      archive.pipe(mWriter);
```

Comment:
I believe every chunk is a buffer, otherwise you would need to add isBuffer check in MemWritable.

Reply:
The `_write` overload on `MemWritable` just pushes the buffers it receives to an array. When we call `get buffer` on the class (after `archiver` has signaled it is done), it uses `Buffer.concat` to merge them all. Then we use `toString()` on the merged buffer when this promise returns.

Does that make sense?

Reply:
I get the functionality, but if you end up piping objects then it wont be a buffer anymore. Its not a huge deal though. Lets not worry too much.

Reply:
Ah like if the input is unexpected? It _should_ always be a string because the inline script field has been validated by the time it reaches this point. I can add a case to throw an error for invalid input at the top of the function, would that make this resilient to the bug you're talking about?

Reply:
Yes, if there is already a validation. I am good with it.